src/Engine/EmitMSIL/Replication.cs

```diff
@@ -133,5 +242,324 @@ private static List<List<ReplicationGuide>> ConstructRepGuides(string[][] replic
+            // TODO_MSIL: Determine final function endpoint here based on fitting runtime args to function parameters
+            return functionEndPoints.FirstOrDefault();
+        }
+
+        private static object ExecWithZeroRI(CLRFunctionEndPoint finalFep, List<CLRStackValue> formalParameters, MSILRuntimeCore runtimeCore)
+        {
+            List<CLRStackValue> coercedParameters = finalFep.CoerceParameters(formalParameters, runtimeCore);
+
+            List<object> args = UnmarshalFunctionArguments(finalFep.FormalParams, coercedParameters, runtimeCore);
+
+            // Testing invoking method without replication
+            object result;
+            if (finalFep.method.IsStatic)
+            {
+                result = finalFep.method.Invoke(null, args.ToArray());
+            }
+            else
+            {
+                result = finalFep.method.Invoke(args[0], args.Skip(1).ToArray());
+            }
+
+            var marshaller = ProtoFFI.CLRDLLModule.GetMarshaler(runtimeCore);
+            CLRStackValue dsRetValue = marshaller.Marshal(result, ProtoFFI.CLRObjectMarshaler.GetProtoCoreType(finalFep.ReturnType), runtimeCore);
+
+            if (!dsRetValue.IsExplicitCall)
+            {
+                // An explicit call requires return coercion at the return instruction
+                dsRetValue = CallSite.PerformReturnTypeCoerce(finalFep, dsRetValue, runtimeCore);
+            }
+
+            var returnVal = marshaller.UnMarshal(dsRetValue, finalFep.ReturnType, runtimeCore);
+            marshaller.OnDispose(dsRetValue);
+
+            return returnVal;
+        }
+
+        private static IList<CLRStackValue> getSubParameters(CLRStackValue o)
+        {
+
+            if (o.IsEnumerable)
+            {
+                return o.Value as IList<CLRStackValue>;
+            }
+            else
+            {
+                return new List<CLRStackValue>() { o };
+            }
+        }
+
+        private static object ExecWithRISlowPath(CLRFunctionEndPoint finalFep, List<CLRStackValue> formalParameters,
+            List<ReplicationInstruction> replicationInstructions, MSILRuntimeCore runtimeCore)
+        {
+            //Recursion base case
+            if (replicationInstructions.Count == 0)
+            {
+                return ExecWithZeroRI(finalFep, formalParameters, runtimeCore);
+            }
+
+            //Get the replication instruction that this call will deal with
+            ReplicationInstruction ri = replicationInstructions[0];
+
+            if (ri.Zipped)
+            {
+                ZipAlgorithm algorithm = ri.ZipAlgorithm;
+
+                //For each item in this plane, an array of the length of the minimum will be constructed
+
+                //The size of the array will be the minimum size of the passed arrays
+                List<int> repIndecies = ri.ZipIndecies;
+
+                //this will hold the heap elements for all the arrays that are going to be replicated over
```

Comment:
This comment doesn't seem to apply anymore.

Reply:
removed